scripts/replace.js

```diff
@@ -20,13 +20,15 @@ for (const filepath of [leafletOptions, debug]) {
   const BACKEND = process.env.OSRM_BACKEND || 'https://router.project-osrm.org'
   const LANGUAGE = process.env.OSRM_LANGUAGE || 'en'
   const DEFAULT_LAYER = process.env.OSRM_DEFAULT_LAYER || 'streets'
+  const MAPBOX_TOKEN = process.env.OSRM_MAPBOX_TOKEN || 'pk.eyJ1IjoibXNsZWUiLCJhIjoiclpiTWV5SSJ9.P_h8r37vD8jpIH1A6i1VRg'
 
   // Edit Leaflet Options
   if (BACKEND) options = options.replace(/http[s]?:\/\/router\.project-osrm\.org/, BACKEND)
   if (LABEL) options = options.replace('Car (fastest)', LABEL)
   if (ZOOM) options = options.replace('zoom: 13', `zoom: ${ZOOM}`)
   if (LANGUAGE) options = options.replace(`language: 'en'`, `language: '${LANGUAGE}'`)
   if (DEFAULT_LAYER) options = options.replace('layer: streets', `layer: ${DEFAULT_LAYER}`)
+  if (MAPBOX_TOKEN) options = options.replace(`mapboxToken: '.*'`, `mapboxToken: '${MAPBOX_TOKEN}'`)
```

Comment:
You can verify if this is working by running the replace script. E.g.
```
OSRM_MAPBOX_TOKEN=foo npm run replace
```
Looks like this needs tweaking as the field is not exported like the other options.

Option 1 - Regexp
```suggestion
  if (MAPBOX_TOKEN) options = options.replace(/mapboxToken = '.*'/, `mapboxToken = '${MAPBOX_TOKEN}'`)
```
Option 2 - Hardcoded token
```suggestion
  if (MAPBOX_TOKEN) options = options.replace(`mapboxToken = 'pk.eyJ1IjoibXNsZWUiLCJhIjoiclpiTWV5SSJ9.P_h8r37vD8jpIH1A6i1VRg'`, `mapboxToken = '${MAPBOX_TOKEN}'`)
```

Reply:
Thanks @mjjbell . I merged the the first option as it does not rely on the hardcoded default token.